lms/djangoapps/courseware/tests/test_transformers.py

```diff
@@ -0,0 +1,158 @@
+        super(GradesBlockTransformerTestCase, self).setUp()
+        password = u'test'
+        self.student = UserFactory.create(is_staff=False, username=u'test_student', password=password)
+        self.client.login(username=self.student.username, password=password)
+
+    def assert_collected_xblock_fields(self, block_structure, usage_key, **expectations):
+        """
+        Given a block structure, a block usage key, and a list of keyword
+        arguments representing XBlock fields, verify that the block structure
+        has the specified values for each XBlock field.
+        """
+        self.assertGreater(len(expectations), 0)
+        for field in expectations:
+            self.assertEqual(
+                expectations[field],
+                block_structure.get_xblock_field(usage_key, field),
+            )
+
+    def assert_collected_transformer_block_fields(self, block_structure, usage_key, transformer_class, **expectations):
+        """
+        Given a block structure, a block usage key, a transformer, and a list
+        of keyword arguments representing transformer block fields, verify that
+        the block structure has the specified values for each transformer block
+        field.
+        """
+        self.assertGreater(len(expectations), 0)
+        for field in expectations:
+            self.assertEqual(
+                expectations[field],
+                block_structure.get_transformer_block_field(usage_key, transformer_class, field),
+            )
+
+    def build_course_with_problems(self, data='<problem></problem>', metadata=None):
+        """
+        Create a test course with the requested problem `data` and `metadata` values.
+
+        Appropriate defaults are provided when either argument is omitted.
+        """
+        metadata = metadata or self.problem_metadata
+        return self.build_course([
+            {
+                u'org': u'GradesBlockTestOrg',
+                u'course': u'GB101',
+                u'run': u'cannonball',
+                u'#type': u'course',
+                u'#ref': u'course',
+                u'#children': [
+                    {
+                        u'metadata': metadata,
+                        u'#type': u'problem',
+                        u'#ref': u'problem',
+                        u'data': data,
+                    }
+                ]
+            }
+        ])
+
+    def test_grades_collected_basic(self):
+
+        blocks = self.build_course_with_problems()
+        block_structure = get_course_blocks(self.student, blocks[u'course'].location, self.transformers)
+
+        self.assert_collected_xblock_fields(
+            block_structure,
+            blocks['problem'].location,
+            weight=self.problem_metadata[u'weight'],
+            graded=self.problem_metadata[u'graded'],
+            has_score=True,
+            due=self.problem_metadata[u'due'],
+        )
+
+    def test_collecting_staff_only_problem(self):
```

Comment:
To verify that the staff_only problem is filtered out for `self.student`, you would need to include the `COURSE_BLOCK_ACCESS_TRANSFORMERS` in the list of transformers sent to `get_course_blocks`.